### DISCUSSION TITLE
Using `tectonic` with `AUCTeX`?

### DISCUSSION THREAD

**Discussion by astrogewgaw** [tectonic-typesetting/tectonic#927] [Q&A]
<div align="justify">

Does anyone have any idea how I can configure `AUCTeX` to use `tectonic`? I am currently trying to setup the LaTeX layer in [**spacemacs**](https://github.com/syl20bnr/spacemacs) which uses `AUCTeX`, and I can't get it to work with `tectonic`. It seems that people have either not tried this at all, or that they haven't posted about it even if they have. I could use some help here, because googling the problem certainly has not helped 😅 .

</div>

**Comment by pkgw** [comment]
Looks like not :-( If you post any specific problems or error messages that you're encountering, we can try to help you work through, though!

**Comment by KaranAhlawat** [comment]
I spent the majority of today setting up, debugging and in general fighting with AucTeX to use tectonic. 
But I'm happy to say I've arrived at a result I think is quite good! 

It offers all the basics for now, such as
1. Compile on save
2. PDF preview in your program of choice
3. Live PDF update
4. Improved completions with LSP

I am for now happy with how this has played out, and would like to help others if possible!
Just don't know what would be the best way to share my config so it's actually visible and not buried in Discussions after a while.

@pkgw Thoughts?

**Comment by pkgw** [comment]
Sounds very cool, and thank you for putting in the work on this! Is your result like a single config file, or a set of steps to go through, or what? But either way, the [book][1] has a section for "how-to guides" where this kind of information could go.

[1]: https://tectonic-typesetting.github.io/book/latest/

**Reply by KaranAhlawat** [reply]
Thank *you* for working on Tectonic! 
My result is essentially a single configuration file, although it could do with a bit of explanation I suppose, so it's not black magic to however uses it.

I'll try to make a PR for the how-to guide section in the book if possible.

**Reply by pkgw** [reply]
Thanks! For posterity, that PR has been filed as #945. Discussion of the specifics to proceed there.
